commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/ext/XMLStreamReaderExt.kt

```diff
@@ -0,0 +1,18 @@
+package ac.uk.ebi.biostd.ext
+
+import javax.xml.stream.XMLStreamReader
+
+/**
+ * Iterates over each XML node loaded by the reader and applies the given function .
+ *
+ * @param function The function to be applied to each element.
+ */
+fun XMLStreamReader.forEach(function: () -> Unit) {
```

Comment:
2 issues in this function.

1. Is not safe (user need to remember to call next and close) otherwise will work incorrectly.
2. Iteration should be over reader object so user does not need to access externally variable.